src/main/java/com/boomi/flow/services/boomi/mdh/records/GoldenRecordRepository.java

```diff
@@ -39,19 +39,31 @@ public void delete(ApplicationConfiguration configuration, String universeId, Li
 
         // TODO: Cleanup everything in this filter block cause it's super ugly
         if (filter != null) {
+            GoldenRecordQueryRequest.Sort sort = new GoldenRecordQueryRequest.Sort();
+
+            // Add component filter to sort
+            if (filter.hasOrderByPropertyDeveloperName() && filter.getOrderByDirectionType() != null) {
+                sort.getFields().add(new GoldenRecordQueryRequest.Sort.Field()
+                        .setFieldId(filter.getOrderByPropertyDeveloperName().toUpperCase())
+                        .setDirection(filter.getOrderByDirectionType().toString())
+                );
+            }
 
+            // Add metadata orderBy[] to sort
             if (filter.hasOrderBy()) {
-                GoldenRecordQueryRequest.Sort sort = new GoldenRecordQueryRequest.Sort();
-
                 for (ListFilter.OrderBy orderBy : filter.getOrderBy()) {
-                    sort.getFields().add(new GoldenRecordQueryRequest.Sort.Field()
-                            .setFieldId(orderBy.getColumnName())
-                            .setDirection(orderBy.getDirection())
-                    );
+                    String currFieldId = orderBy.getColumnName().toUpperCase();
+                    boolean fieldIdExists = sort.getFields().stream().anyMatch(item -> item.getFieldId().equals(currFieldId));
```

Comment:
I think we only want to ignore this field for filter if it was added in the if block of [line 45](https://github.com/manywho/service-mdh/pull/56/files#diff-6f276202a3e4f63532ceaa67e1d9c353R45). If the user add an orderBy with the same field two times we should send the request to MDH and let MDH decide if it is right or not. 
You could in that case also get rid of the stream search and use "getOrderByPropertyDeveloperName"  to check if the field should be ignored.

Reply:
Will update this logic and remove the stream :+1: